nim/static/game_operations.js

```diff
@@ -13,18 +13,44 @@ function newGame() {
     }),
     contentType: "application/json; charset=utf-8",
     dataType : 'json',
-    success: updatePage
+    success: function( json ) { updatePage( json.state ) }
   });
 };
 
 function updateGameState() {
-  $.getJSON( "/update", updatePage);
+  $.ajax({
+    type:"POST",
+    url: "/update",
+    data: JSON.stringify({
+      "pile": parseInt($('#pile').val()) - 1,
+      "stones": parseInt($('#stones').val()),
+    }),
+    contentType: "application/json; charset=utf-8",
+    dataType : 'json',
+    success: applyMoves
+  });
+};
+
+function applyMoves( json ) {
+  if ( json.playerState ) {
+    updatePage( json.playerState )
+  }
+
+  if ( json.botState && !$('#delay').is(":checked") ) {
+    setTimeout(function() { updatePage( json.botState ) }, 100);
+  }
+
+  else if ( json.botState ) {
+    setTimeout(function() { updatePage( json.botState ) }, 1000);
+  }
```

Comment:
I think you should be able to factorize this. maybe:
```
  if ( json.botState ) {
    setTimeout(function() { updatePage( json.botState ) }, $('#delay').is(":checked")? 1000: 100);
  }
```

Reply:
Pushing refactor. No longer applicable.